src/dnd/DragArea.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import { DndContext } from './DndContext';
+
+class DragArea extends Component {
+  state = {
+    dragStart: false,
+    draggable: null,
+    x: 0,
+    y: 0,
+    lx: 0,
+    ly: 0,
+    deg: 0,
+  }
+
+  mouseDown = (e) => {
+    this.setState({
+      dragStart: true,
+      ly: e.clientY,
+      lx: e.clientX,
+    });
+  }
+
+  mouseUp = () => {
+    this.transformElement(this.state.draggable, 0, 0, 0);
+    this.setState({
+      dragStart: false,
```

Comment:
this call restores state to its initial value

Let's reuse it with help of function? Smth like getInitialState ?